### PR TITLE
Refactor `ClockMap`

### PR DESCRIPTION
**Pull request by ffuugoo**
Tracked in: <https://github.com/qdrant/qdrant/issues/3477>

Some minor `ClockMap` improvements:

- simplify documentation and implementation
- remove accidentally committed debug logging
- restructure the code, so that we still accept all operations, but without modifying `ClockMap` itself

### All Submissions:

* [x] Contributions should target the `dev` branch. Did you create your branch from `dev`?
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [ ] Does your submission pass tests?
2. [x] Have you formatted your code locally using `cargo +nightly fmt --all` command prior to submission?
3. [x] Have you checked your code using `cargo clippy --all --all-features` command?

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your core changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by generall** [comment]
nitpick: can't we also replace `store` function of `ClockMap` with `atomic_save_json`?

**Comment by ffuugoo** [comment]
> nitpick: can't we also replace `store` function of `ClockMap` with `atomic_save_json`?

Sure. I didn't know we have it. :D

**Review by generall — APPROVED** [review]

File: lib/collection/src/shards/local_shard/clock_map.rs [hunk 1]
```diff
@@ -49,76 +48,60 @@ impl ClockMap {
...
+        // There are two special cases:
+        // - we *always* accept operations with `clock_tick = 0`, and also *always* update their `clock_tick`
+        // - and we *always* accept operations with `force = true`, but *never* update their `clock_tick`
 
-        let _operation_accepted = clock_updated || clock_tag.clock_tick == 0 || clock_tag.force;
-        let update_tag = (!clock_updated || clock_tag.clock_tick == 0) && !clock_tag.force;
-
-        if _operation_accepted && !update_tag {
-            log::trace!("Accepting clock tag {clock_tag:?} (current tick: {current_tick})");
+        if clock_tag.force {
```
Comment (timvisee) [inline lib/collection/src/shards/local_shard/clock_map.rs:70]: Don't we want to advance with force enabled too?

Comment (ffuugoo) [reply lib/collection/src/shards/local_shard/clock_map.rs:70]: We do advance (at the very top of the method :D), we just don't update the tag.

File: lib/collection/src/shards/local_shard/clock_map.rs [hunk 2]
```diff
@@ -181,11 +148,12 @@ pub enum Error {
...
-impl From<atomicwrites::Error<Error>> for Error {
-    fn from(err: atomicwrites::Error<Error>) -> Self {
+impl From<file_operations::Error> for Error {
+    fn from(err: file_operations::Error) -> Self {
         match err {
-            atomicwrites::Error::Internal(err) => err.into(),
-            atomicwrites::Error::User(err) => err,
+            file_operations::Error::Io(err) => err.into(),
+            file_operations::Error::SerdeJson(err) => err.into(),
+            _ => unreachable!(),
```
Comment (ffuugoo) [inline lib/collection/src/shards/local_shard/clock_map.rs:156]: Is this good enough, or should I convert these to `io::Error` still?

File: lib/collection/src/shards/local_shard/clock_map.rs (hunk 1 above)
Comment (timvisee) [reply lib/collection/src/shards/local_shard/clock_map.rs:60]: You're right! GitHub diffs made it hard to read :smiling_face_with_tear:

File: lib/collection/src/shards/local_shard/clock_map.rs (hunk 2 above)
Comment (timvisee) [reply lib/collection/src/shards/local_shard/clock_map.rs:156]: Fine by me :+1:

**Pull request merged**